python/cudf/cudf/tests/test_orc.py

```diff
@@ -1780,3 +1780,16 @@ def test_orc_columns_and_index_param(index, columns):
         )
     else:
         assert_eq(expected, got, check_index_type=True)
+
+
+def test_orc_writer_cols_as_map_type():
```

Comment:
Would be good to add a test where the map column is a field of a struct column to test the option with nested column names.
Also might be good to include another list column in the dataframe, and write in as a list, to check that the flag did not somehow apply to all columns.

Reply:
Added all the scenarios listed here as part of test coverage 👍